DangoPlop/Assets/Scripts/PowerupDoubleShot.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PowerupDoubleShot : PowerupParent{
+	public float limitDoubleShotAmmo;
+
+	public override void HandlePowerupAction(float powerupLastingTime){
+		powerupMaster.startPowerupAction (PowerupType.Double, powerupLastingTime);
+		playerController.currentDoubleShotAmmo = limitDoubleShotAmmo;
```

Comment:
the powerup class needs to communicate with the powerupmaster to communicate with the player. the powerup does not have a player. so it shouldn't have access to the player.